### ISSUE TITLE
The abort request doesn't exit and display the error loading message

### ISSUE THREAD

**Issue by laudeco** [select2/select2#4781]
## Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate
- [X] The issue still exists against the latest `master` branch of Select2
- [X] This is not a usage question (Those should be directed to the [community](https://select2.github.io/community.html))
- [X] I have attempted to find the simplest possible steps to reproduce the issue
- [ ] I have included a failing test as a pull request (Optional)

## Steps to reproduce the issue

1. Click and search something in the search box
2. During the search type something else
3. The abort message is present waiting the response of the latest query

## Expected behavior and actual behavior

When I follow those steps, I see...
The errorLoading message

I was expecting...
The searching message

## Environment

Browsers

- [X] Google Chrome : 56.0
- [ ] Mozilla Firefox
- [ ] Internet Explorer

Operating System

- [X] Windows
- [X] Mac OS X
- [ ] Linux
- [ ] Mobile

Libraries

- jQuery version: 1.11.2
- Select2 version: 4.0.3

## Isolating the problem

- [X] This bug happens [on the examples page](https://select2.github.io/examples.html)
- [X] The bug happens consistently across all tested browsers
- [X] This bug happens when using Select2 without other pluigns
- [ ] I can reproduce this bug in [a jsbin](https://jsbin.com/)

## Causes
In the select.full.js file on line 3472 of 4.0.3 version.

```
, function () {
        // Attempt to detect if a request was aborted
        // Only works if the transport exposes a status property
        if ($request.status && $request.status === '0') {
          return;
        }

        self.trigger('results:message', {
          message: 'errorLoading'
        });
      });
```

The status code is always an _int_ in my browser so when it's an abort request the status is _0_ . It means that it will never rich the _return;_ .

**Comment by laudeco** [comment]
No news about this ?

**Comment by jpic** [comment]
Couldn't reproduce, tried:

- "Loading remote data example",
- in the Network tab of webkit inspector, set Throttling: GPRS,
- type "django"
- replace it with "node" very fast

I have just seen the "searching" message, not the error message.

**Comment by jpic** [comment]
Is this a duplicate of #4355 ?

**Comment by medmondson** [comment]
I'm on Chromium 56 using the "Loading remote data example":

Start by typing **gal**

![1](https://cloud.githubusercontent.com/assets/1955888/23220537/35e589be-f91a-11e6-9ec7-38ae3f20d67c.png)

**Before the results are loaded** type **d**. Error displays.

![2](https://cloud.githubusercontent.com/assets/1955888/23220542/389d49f8-f91a-11e6-9edc-ef59e27d2c70.png)

Wait a short period results show

![3](https://cloud.githubusercontent.com/assets/1955888/23220544/399ae324-f91a-11e6-9af3-3317be3278eb.png)

**Comment by alexweissman** [comment]
This should be fixed as of 4.0.6-rc.0.

**Comment by burnchar** [comment]
Confirmed fixed in 4.0.6-rc.1 (did not test rc.0). Thanks, Alex!
